api/types.go

```diff
@@ -157,6 +157,32 @@ func (tr Timerange) IsValid() bool {
 		tr.Start <= tr.End
 }
 
+func rounder(x, r int64) int64 {
+	if x < 0 {
+		return -rounder(-x, r)
+	}
+	// This performs a round
+	return (x + r/2) / r * r
```

Comment:
Should leave a comment explaining what you're trying to do here. It doesn't have to be too detailed, but at first glance the `/ r * r` isn't quite clear.